internal/transport/http2_server.go

```diff
@@ -946,7 +950,7 @@ func (t *http2Server) WriteHeader(s *Stream, md metadata.MD) error {
 	}
 	if err := t.writeHeaderLocked(s); err != nil {
```

Comment:
GH won't let me comment on the right line, but on line 981, there's no guarantee this is a status error.  I think it's always `ErrConnClosing`, in fact.  Let's make that `return status.Convert(err).Err()` -- it's not really efficient or ideal, but it's only in an error condition so it should be okay.  This means a closed connection will result in UNKNOWN, but since I'm pretty sure using statuses at all here is a bad idea in the first place, that should be fine.  I think I'll separately change all the documentation on `ServerStream` to recommend against relying upon the returned errors' status codes.

Reply:
Are you suggesting changing the error inside `writeHeaderLocked` to be `status.Convert(err).Err()` or changing the handling of the return value to be `status.Convert(err).Err()`?

I changed the outside error, as that's the place where we converted it to `status.Internal` previously. Let me know if I understood wrong.

Reply:
Either way is probably fine.  There is only one other usage of `writeHeaderLocked`, in `WriteStatus`, and the ultimate user of that just logs the result.